metrics/cli/docs/features/update/06_update_feature.md

```diff
@@ -165,4 +175,139 @@ Having considered all the complex points and edge cases, we conclude that the `U
+
+Let's take a closer look at the structure of the configuration file parsing. The below subsections examine the related classes and their relationships we should implement.
+
+#### UpdateConfig
+
+The `UpdateConfig` is a model that represents the `YAML` update config. The `UpdateConfig` class should aggregate the following classes:
+
+- the `FirebaseConfig` class - represents the `firebase` attribute of the update `YAML` config;
+- the `SentryConfig` class - represents the `sentry` attribute of the update `YAML` config.
+
+_**Note**: The `SentryConfig` field should be `null` if the `sentry` parameter of the update `YAML` config is empty._
+
+Let's review the detailed structure of the `UpdateConfig` by looking at the following class diagram:
+
+![Update config class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/Flank/flank-dashboard/raw/update_feature_design/metrics/cli/docs/features/update/diagrams/update_config_class_diagram.puml)
+
+#### UpdateConfigParser
+
+The `UpdateConfigParser` is a class uses to parse the `YAML` configuration file data to the [`UpdateConfig`](#updateconfig) model. The `UpdateConfigParser` should use the [`yaml_map`](https://github.com/Flank/flank-dashboard/tree/master/yaml_map) package for parsing purposes, as was analyzed in the [YAML map parser section](#yaml-map-parser).
+
+#### UpdateConfigFactory
+
+The `UpdateConfigFactory` is a class uses to create the [`UpdateConfig`](#updateconfig) using the [`UpdateConfigParser`](#updateconfigparser). This class should contain only one method `create()` that takes the path to the configuration file as an input parameter.
+
+The following class diagram demonstrates the structure of the classes required for the `YAML` config parsing process:
+
+![Config YAML parser class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/Flank/flank-dashboard/raw/update_feature_design/metrics/cli/docs/features/update/diagrams/config_yaml_parser_class_diagram.puml)
+
+### Redeploy Metrics components
+
+The following section describes the classes required for the redeploy process.
+
+#### UpdateCommand
+
+The `UpdateCommand` is the `Metrics CLI` command that is responsible for the redeploying the last version of the Metrics Web Application to the Firebase using the data from the [`UpdateConfig`](#updateconfig).
+
+The `UpdateCommand` algorithm performs the following steps:
+
+1. Parse the YAML config.
+2. Checkout the Metrics Web Application source code.
+3. Build the Metrics Web Application.
+4. Configure Sentry using data from the config.
+5. Redeploy the Firebase components like Firestore rules and Firebase Functions using data from the config.
+6. Redeploy the Metrics Web Application to the Firebase Hosting using data from the config.
+7. Cleanup the created directories, etc.
```

Comment:
It looks like that it performs less steps actually as it uses `Updater` to actually update redeploy. Should we leave here something simpler - like: "Uses Updater to ...." and then in `Updater` section specify what redeployment logic from this list is actually performed in Updater. :readable: